### PR TITLE
add sfu voice store

### PR DESCRIPTION
**Pull request by gonpombo8**
# What? 
...

# Why? 
...

### CODE REVIEW

File: packages/shared/voice/sagas/signalConnection.ts [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+    if (voiceEnabled || true) { // TODO remove this
+      const { client, signal }: SignalConnection = yield call(createSignalConnection)
+      const socketChannel: EventChannel<VoiceActions> = yield call(createSocketChannel, { client, signal })
+
+      yield put(voiceInitialized(signal, client))
+      yield call(joinRoom)
+
+      while (true) {
+        try {
+          const payload: VoiceActions = yield take(socketChannel)
```
Comment (menduz) [inline packages/shared/voice/sagas/signalConnection.ts:100]: payload can be an error due to

```ts
    // If signal closes, then throw a new error.
    // The try/catch wrapper would do the reconnect workflow
    signal.onclose = (event) => {
      emit(new Error(event.type))
    }
```

is that intended?

File: packages/shared/voice/sagas/signalConnection.ts [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+          yield put(payload)
+        } catch (error) {
+          // tslint:disable-next-line: no-console
+          console.log('Channel error:', error)
```
Comment (menduz) [inline packages/shared/voice/sagas/signalConnection.ts:104]:
```suggestion
          sharedLogger.log('Channel error:', error)
```

File: packages/shared/voice/sagas/signalConnection.ts [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+          socketChannel.close()
+          throw error
+        }
+      }
+    }
+  } catch (error) {
+    // tslint:disable-next-line: no-console
+    console.log('Voice Saga:', error)
```
Comment (menduz) [inline packages/shared/voice/sagas/signalConnection.ts:112]:
```suggestion
    sharedLogger.log('Voice Saga:', error)
```

File: packages/shared/voice/sagas/signalConnection.ts [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+  reconnectVoice
+} from '../actions'
+import { VoiceState } from '../types'
+import { joinRoom } from './joinRoom'
+
+type SignalConnection = Required<Pick<VoiceState, 'signal' | 'client'>>
+
+// TODO remove constants
+const ONE_MIMNUTE = 1000 * 60
+const URL = 'wss://test-sfu.decentraland.zone/ws'
```
Comment (menduz) [inline packages/shared/voice/sagas/signalConnection.ts:23]: please use the .services SFU, .zone doesn't work AFAIK

**Review by menduz — CHANGES_REQUESTED** [review]
Is this ready for review? does it handle island changes?

File: packages/shared/voice/sagas/signalConnection.ts (hunk 1 above)
Comment (gonpombo8) [reply packages/shared/voice/sagas/signalConnection.ts:100]: that emit new Error is handled by the `catch block`, so paylod can only be VoiceActions.

**Comment by gonpombo8** [comment]
> Is this ready for review? does it handle island changes?
@menduz nope, its missing the entire VoiceContext part, but all the redux part its done. 

The island changes is handled here https://github.com/decentraland/kernel/pull/74/files#diff-f415fbae518cdf80eca665bfd3091096b819ad1d01bdc643c6cab48a3c6a32f1R21.

Also there is a bug or something i dont understand that leaving the current room its not working, so i had to do some trick (disconnect and connect the ws 😢). Im still looking for a way to resolve this.

**Pull request closed without merging**